dlme_airflow/drivers/oai_xml.py

```diff
@@ -21,15 +23,24 @@ class OaiXmlSource(intake.source.base.DataSource):
     partition_access = True
 
     def __init__(
-        self, collection_url, metadata_prefix, set=None, dtype=None, metadata=None
+        self,
+        collection_url,
+        metadata_prefix,
+        set=None,
+        wait=0,
+        dtype=None,
+        metadata=None,
     ):
         super(OaiXmlSource, self).__init__(metadata=metadata)
         self.collection_url = collection_url
         self.metadata_prefix = metadata_prefix
         self.record_limit = self.metadata.get("record_limit", None)
         self.record_count = 0
         self.set = set
-        self._collection = Sickle(self.collection_url)
+        self.wait = wait
+        self._collection = Sickle(
+            self.collection_url, iterator=make_iterator(self.wait)
```

Comment:
Instead of letting Sickle use the default iterator class (OAIItemIterator) we make our own using the`make_iterator()` function and whatever value of `wait` was used.